### PR TITLE
Iterface segregation, type-safe casting of context[transaction]

### PR DESCRIPTION
**Pull request by p-kozlowski**
This PR consists of two parts:
1. refactoring: extracting Contig and UUID from Core into separate interfaces (to facilitate mocking in unit tests)
2. proper casting of context[transaction]

It may happen that context[transaction] is a transactionCommitInformer, which does not fulfill goext.ITransaction.

It must be a part of public (goext) interface because Within() relies on this functionality. It may be interesting to implement Within() using dependency injection, but that complicates testing of plugins when mocks are used.

### CODE REVIEW

File: extension/goext/util.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package goext
```
Comment (morrisson) [inline extension/goext/util.go]: license

File: extension/goplugin/util.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package goplugin
```
Comment (morrisson) [inline extension/goplugin/util.go]: license

File: extension/goext/util.go (hunk 1 above)
Comment (p-kozlowski) [reply extension/goext/util.go]: added

File: extension/goplugin/util.go (hunk 2 above)
Comment (p-kozlowski) [reply extension/goplugin/util.go]: added

**Comment by p-kozlowski** [comment]
@morrisson importing parts of Gohan (except goext) in plugins will work for same cases (like here), but in general we'd like to use Dependency Insertion and Dependency Injection for a number of reasons:

1. resulting binary size increases
2. you may encounter import cycles
3. compilation of plugins may fail

Therefor, I'd like to keep consistency and follow the rule for all interfaces.

**Pull request merged**